translate/convert/factory.py

```diff
@@ -64,9 +61,6 @@ def __str__(self):
             msg += ' with template %s' % (self.templ_ext)
         return msg
 
-    def __unicode__(self):
-        return unicode(str(self))
```

Comment:
By removing these `__unicode__()` methods we are now relying on `__str__()` ones. And those don't seem to convert to Unicode strings on Python 2, so perhaps the output of `__str__()` must also be adjusted to be converted using `six.text_type`.